C error when assigning generic fixed size array to a variable

### Describe the bug

```v
module main

fn init_type[T]() {
	_ := T{}
}

fn main() {
	init_type[[3]int]()
}
```
gives:
```
/tmp/v_1002/src.5677094858549572046.tmp.c:12624: error: '{' expected (got ";")
```

### Reproduction Steps

Try compiling the above code

### Expected Behavior

Compile successfully

### Current Behavior

```
/tmp/v_1002/src.5677094858549572046.tmp.c:12624: error: '{' expected (got ";")
```

### Possible Solution

_No response_

### Additional Information/Context

Compiling with `v -cc clang` makes it compile successfully
Issue very similar to #19872

### V version

V 0.4.3 e9258c2

### Environment details (OS name and version, etc.)

```yaml
V full version: V 0.4.3 4628fe7.e9258c2
OS: linux, Linux version 6.5.11-300.fc39.x86_64 (mockbuild@d23353abed4340e492bce6e111e27898) (gcc (GCC) 13.2.1 20231011 (Red Hat 13.2.1-4), GNU ld version 2.40-13.fc39) #1 SMP PREEMPT_DYNAMIC Wed Nov  8 22:37:57 UTC 2023
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz

getwd: /home/user/Projects/v-testing/webview
vexe: /home/user/.local/lib/v/v
vexe mtime: 2023-11-16 16:24:37

vroot: OK, value: /home/user/.local/lib/v
VMODULES: OK, value: /home/user/.vmodules
VTMP: OK, value: /tmp/v_1002

Git version: git version 2.41.0
Git vroot status: weekly.2023.45.1-54-ge9258c2a
.git/config present: true

CC version: cc (GCC) 13.2.1 20231011 (Red Hat 13.2.1-4)
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5247,7 +5247,7 @@ fn (mut g Gen) return_stmt(node ast.Return) {
 					g.writeln('{0};')
 					if node.exprs[0] is ast.Ident {
 						g.write('memcpy(${tmpvar}.ret_arr, ${g.expr_string(node.exprs[0])}, sizeof(${g.typ(node.types[0])})) /*ret*/')
-					} else if node.exprs[0] is ast.ArrayInit {
+					} else if node.exprs[0] in [ast.ArrayInit, ast.StructInit] {
 						tmpvar2 := g.new_tmp_var()
 						g.write('${g.typ(node.types[0])} ${tmpvar2} = ')
 						g.expr_with_cast(node.exprs[0], node.types[0], g.fn_decl.return_type)

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -48,8 +48,9 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 	if mut sym.info is ast.Struct {
 		is_anon = sym.info.is_anon
 	}
+	is_array := sym.kind in [.array_fixed, .array]
 
-	if !g.inside_cinit && !is_anon {
+	if !g.inside_cinit && !is_anon && !is_array {
 		g.write('(')
 		defer {
 			g.write(')')
@@ -88,7 +89,9 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 		if g.table.sym(node.typ).kind == .alias && g.table.unaliased_type(node.typ).is_ptr() {
 			g.write('&')
 		}
-		if is_multiline {
+		if is_array {
+			g.write('{')
+		} else if is_multiline {
 			g.writeln('(${styp}){')
 		} else {
 			g.write('(${styp}){')

--- a/vlib/v/tests/generics_fixed_array_assign_test.v
+++ b/vlib/v/tests/generics_fixed_array_assign_test.v
@@ -1,0 +1,8 @@
+fn init_type[T]() {
+	_ := T{}
+}
+
+fn test_main() {
+	init_type[[3]int]()
+	assert true
+}

